### PR TITLE
feat: add select tickets page

### PR DESCRIPTION
**Pull request by SamuelBenkeCalabresi**
feat: add select tickets page

https://github.com/planvine/lineup-js-sdk/assets/17012108/d0509a2e-9beb-4f86-90e7-fda3debeab11

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204496466135305

### CODE REVIEW

File: apps/lineup-mercury/src/app/components/Pricing/usePrincingRows.tsx
Comment (nananathanh) [inline apps/lineup-mercury/src/app/components/Pricing/usePrincingRows.tsx:1]: Typo in the filename `usePrincingRows` -> `usePricingRows`

File: apps/lineup-mercury/src/app/components/Pricing/usePrincingRows.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  priceMap: IPriceMap
+  variant: IPriceVariant
+  price: IPrice
+  priceAdjuster?: IPriceAdjuster
+  externalPriceAdjuster?: number
+  quantityInBasket: number
+  description?: string
+}
+
+const usePrincingRows = ({ pricing }: { pricing: IPriceMap[] }) => {
```
Comment (nananathanh) [inline apps/lineup-mercury/src/app/components/Pricing/usePrincingRows.tsx:14]: Typo in the component too

File: apps/lineup-mercury/src/app/components/ProductItem/ProductItem.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+          style={{
+            margin: '0',
+          }}
+        >
+          {product.title}
+        </Heading>
+        <Text>{product.description}</Text>
+        <Box direction="row" align="center" justify="between">
+          <Box border pad="xsmall">
+            10.00
```
Comment (nananathanh) [inline apps/lineup-mercury/src/app/components/ProductItem/ProductItem.tsx:23]: Is this meant to be a hardcoded value?

File: apps/lineup-mercury/src/app/components/Sidebar/SidebarContent/Perfomance.tsx [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+  // Handlers
+  const handleMonthChange = (month: Date) => {
+    const firstVisibleDay = startOfISOWeek(startOfMonth(month))
+    const lastVisibleDay = endOfISOWeek(endOfMonth(month))
+    setStartDate(firstVisibleDay)
+    setEndDate(lastVisibleDay)
+  }
+
+  const handlePerformanceClick = (timeslot: TimeSlot) => {
+    // setSelectedDate(timeslot.startDate)
```
Comment (nananathanh) [inline apps/lineup-mercury/src/app/components/Sidebar/SidebarContent/Perfomance.tsx:69]: Should this be included? If not best to remove the line than keeping in commented code

File: apps/lineup-mercury/src/app/components/Sidebar/SidebarContent/Perfomance.tsx [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+    navigate(`/event/${eventId}/performance/${timeslot.id}`)
+  }
+
+  const handlePerformancesClick = (timeslots: TimeSlot[]) => {
+    const performances = timeslots as IPerformance[]
+    // setSelectedDate(performances[0].startDate)
```
Comment (nananathanh) [inline apps/lineup-mercury/src/app/components/Sidebar/SidebarContent/Perfomance.tsx:75]: same as above

File: apps/lineup-mercury/src/app/components/Sidebar/SidebarContent/Product.tsx [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+  // Handlers
+  const handleMonthChange = (month: Date) => {
+    const firstVisibleDay = startOfISOWeek(startOfMonth(month))
+    const lastVisibleDay = endOfISOWeek(endOfMonth(month))
+    setStartDate(firstVisibleDay)
+    setEndDate(lastVisibleDay)
+  }
+
+  const handlePerformanceClick = (timeslot: TimeSlot) => {
+    // setSelectedDate(timeslot.startDate)
```
Comment (nananathanh) [inline apps/lineup-mercury/src/app/components/Sidebar/SidebarContent/Product.tsx:71]: same

File: apps/lineup-mercury/src/app/containers/PageLayout/PageLayout.tsx [hunk 6]
```diff
@@ -11,23 +11,32 @@ export function PageLayout({ children, ...props }: PageLayoutProps) {
...
...F1zNTlUezA2LC4HfomlZ90FO3mjmkGWa2l5LsFhBX+nzVw6An1FJHqEFGKeWsj6N1DVlSIyBUtysQlwv4hcjkUSx20xa4FKC2AP4jcxlpNcsR/wWsxnJxGT1CFM/U4M/31BVV8UkbsxAQwQkT6q+kEZ/SgHmwEXisgKbBTZB7MBwOyml1L4fYnfA3AiZuwV4/fA3th97wlcge2buA74hyZMHb9ByjN0PtCrDBsgyyuLDZD0+lkCfwD2S1TgtPDe7pghpMBFGZ6to2m2AfqXYQPEvZYAAx38tGvujh1IsaCIMws4Oq3/lR4BPsKs0eIhTIHTyZ5g8XMsNr7Y2ldszSEOw0Lbi4D3RaQP9ih4D7PKjxeRy7U6W72jL3tDmkfWZ4Hfarbw8rhrBoh9lqvqF1gW8bHAWdg96I7d69tFZH9VnRHf68qOAFMSyk2lirMAbBXsy8Bfg01dl2JfyiqajabDqzQCfAz0AQZhhqeSsDBWgl/WLCChvu2xjCfRSHBpUvlK+wKStpiVs/2sXY7TMYZgz+GorY0ww7MLzZa9UL1j2FYDn6jqZGx7Gdgi1IiM/HZhSTczROQ4ERlY+J7aEvDFBW8ljrqVfgRUCgI0iUipqaMAqgW7gkL27ugwh4+xZeDibWOjsJNADxWRbdSOlKt4n7GR42LgCGAr4BIReURVs5xl3F5ESj0CBEDtzAN7Q2Q4ltRiqYicgS15R9zCE08Tr3N9FUBfmheOirEWO7+nMLPWgTTPgyep6sRiUlgePgAbJY7HLOaqQFXniMjvgL9j6W3GiMjRqro6gdaH5Gs+DVvXiNCEjXRdsW3tL4rIdOxReEgos5q0vMYxz5F/0mzUbJnBBphF+vP9lVBmWkKZl0m3iN8COhfx7iz4fL+YurfAVjajlcHOMeWuKahr1wzP3GdC2U8xP0f0fhPrWvjnp/CTXm8CnUpwfwMsi+GsBEam9T9uBHgZU9Zi0taSbah9FHvWJK08PYkZle+IiGi4giI8S/N0Jg4TtMAlG45wifowM/S9BVR1voiMxrxna4GtsdlBMaYDj4Q6Y93IBZiKzTQ+xX5xUXtrROQC4I+YGPYUka7a0jUd8eOuWYBbtcTMRVUvE5GnsBnWftjotgxbixmnqqlpbWt+c2iCmNY7VLOvItIBM3pXlhJLHP4PPxS25VRvPCoAAAAASUVORK5CYII="
```
Comment (nananathanh) [inline apps/lineup-mercury/src/app/containers/PageLayout/PageLayout.tsx:18]: Are these images meant to be constant?

File: apps/lineup-mercury/src/app/containers/Product/SelectProducts.tsx [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+          onClick={() => {
+            navigate(`/`)
+          }}
+        >
+          <Box direction="row" align="center" gap="xsmall">
+            <Previous />
+            <Text>Back</Text>
+          </Box>
+        </Button>
+        <Text>Time left to book:</Text>
```
Comment (nananathanh) [inline apps/lineup-mercury/src/app/containers/Product/SelectProducts.tsx:33]: Is the time also meant to be here?

File: apps/lineup-mercury/src/app/containers/PageLayout/PageLayout.tsx (hunk 6 above)
Comment (SamuelBenkeCalabresi) [reply apps/lineup-mercury/src/app/containers/PageLayout/PageLayout.tsx:18]: no @nananathanh this doesn't have an API yet

File: apps/lineup-mercury/src/app/components/ProductItem/ProductItem.tsx (hunk 2 above)
Comment (SamuelBenkeCalabresi) [reply apps/lineup-mercury/src/app/components/ProductItem/ProductItem.tsx:23]: no @nananathanh this needs to be continued

File: apps/lineup-mercury/src/app/containers/Product/SelectProducts.tsx (hunk 7 above)
Comment (SamuelBenkeCalabresi) [reply apps/lineup-mercury/src/app/containers/Product/SelectProducts.tsx:33]: yes @nananathanh this needs to be continued

File: apps/lineup-mercury/src/app/components/ProductItem/ProductItem.tsx (hunk 2 above)
Comment (nananathanh) [reply apps/lineup-mercury/src/app/components/ProductItem/ProductItem.tsx:23]: I think we can merge in without it and Add once we come back to Products

File: apps/lineup-mercury/src/app/containers/Product/SelectProducts.tsx (hunk 7 above)
Comment (nananathanh) [reply apps/lineup-mercury/src/app/containers/Product/SelectProducts.tsx:33]: Similar on Product stuff

**Review by nananathanh — APPROVED** [review]

**Pull request merged**